build_tools/github_actions/github_actions_utils.py

```diff
@@ -157,6 +157,32 @@ def gha_query_workflow_run_information(github_repository: str, workflow_run_id:
     return workflow_run
 
 
+def gha_query_last_successful_workflow_run(
+    github_repository: str, workflow_name: str, branch: str = "main"
+) -> dict | None:
+    """Find the last successful run of a specific workflow on the specified branch.
+
+    Args:
+        github_repository: Repository in format "owner/repo"
+        workflow_name: Name of the workflow file (e.g., "ci_nightly.yml")
+        branch: Branch to filter by (defaults to "main")
```

Comment:
This is useful enough to be sent as a standalone PR.

The code should have unit tests (can use mocked data), and we may want ways to filter the runs further (e.g. by only `schedule` trigger or `nightly` release type (excluding `dev`)